cedar-policy-core/src/extensions/ipaddr.rs

```diff
@@ -242,15 +242,45 @@ fn contains_at_least_two(s: &str, c: char) -> bool {
     let idx = s.find(c);
     match idx {
         Some(i) => {
-            // PANIC SAFETY `i` is guaranteed to be < `s.len()`, so this won't panic
+            // For this slicing operation not to panic, two preconditions must be met:
+            // 1) `i + 1` must be <= `s.len()`
+            //     This is met because `i` comes from `s.find()` meaning it must be < `s.len()`
+            // 2) `i + 1` must be a character boundary
```

Comment:
```suggestion
            // 2) `i + c.len_utf8()` must be a character boundary
```

and this also depends on `i` being the character boundary where  `c` was found